packages/gatsby/src/query/index.js

```diff
@@ -176,37 +57,14 @@ const createStaticQueryJob = (state, queryId) => {
   const component = state.staticQueryComponents.get(queryId)
   const { hash, id, query, componentPath } = component
   return {
-    id: hash,
+    id: queryId,
```

Comment:
Without this change `PAGE_QUERY_RUN` action receives `hash` in `path` property. But we store static query data by id and can't invalidate it. This also seems more consistent (I added another fix for frontend part to actually use hash).